src/main/scala/workflow/WorkflowUtils.scala

```diff
@@ -0,0 +1,124 @@
+package workflow
+
+import scala.collection.mutable.ArrayBuffer
+
+
+object WorkflowUtils {
+  def instructionsToPipeline[A, B](instructions: Seq[Instruction]): Pipeline[A, B] = {
+    val nodes = new ArrayBuffer[Node]()
+    val dataDeps = new ArrayBuffer[Seq[Int]]()
+    val fitDeps = new ArrayBuffer[Option[Int]]()
+    val instructionIdToNodeId = scala.collection.mutable.Map.empty[Int, Int]
+    instructionIdToNodeId.put(Pipeline.SOURCE, Pipeline.SOURCE)
+
+    for (instruction <- 0 until instructions.length) {
+      instructions(instruction) match {
+        case est: EstimatorNode => Unit
+        case transformer: TransformerNode => Unit
+        case source: SourceNode => {
+          instructionIdToNodeId.put(instruction, nodes.length)
+          nodes.append(source)
+          dataDeps.append(Seq())
+          fitDeps.append(None)
+        }
+        case TransformerApplyNode(transformer, inputs) => {
+          instructions(transformer) match {
+            case transformerNode: TransformerNode => {
+              instructionIdToNodeId.put(instruction, nodes.length)
+              nodes.append(transformerNode)
+              dataDeps.append(inputs.map(instructionIdToNodeId.apply))
+              fitDeps.append(None)
+            }
+            case estimatorFitNode: EstimatorFitNode => {
+              instructionIdToNodeId.put(instruction, nodes.length)
+              // TODO: Get more reasonable label...
+              nodes.append(new DelegatingTransformerNode("Estimator fit"))
```

Comment:
can we take the .toString() of the parent node and label it `s"Fit[${parent.toString}]"` or something?